src/client/preview/story_store.js

```diff
@@ -11,6 +11,10 @@ export default class StoryStore {
   }
 
   addStory(kind, name, fn) {
+    if (this.hasStory(kind, name)) {
```

Comment:
I like to do this check here instead of in the store.
See: https://github.com/evgenykochetkov/react-storybook/blob/01f79bc1dc96a414c89fbcd2b568894f6eebff9b/src/client/preview/client_api.js#L48